lib/Cake/Controller/Component/Auth/BasicAuthenticate.php

```diff
@@ -82,8 +82,17 @@ public function authenticate(CakeRequest $request, CakeResponse $response) {
  * @return mixed Either false or an array of user information
  */
 	public function getUser(CakeRequest $request) {
-		$username = env('PHP_AUTH_USER');
-		$pass = env('PHP_AUTH_PW');
+		$username = '';
+		$pass = '';
+		if (isset($_SERVER['PHP_AUTH_USER'])) {
```

Comment:
@ADmad why is the `if...else` not necessary? IMHO it's much more efficient to check a single `isset` than calling `env`.